components/org.wso2.carbon.identity.keyrotation/src/main/java/org.wso2.carbon.identity.keyrotation/config/KeyRotationConfig.java

```diff
@@ -0,0 +1,412 @@
+    private String newSecretKey;
+    private String newISHome;
+    private String oldIdnDBUrl;
+    private String oldIdnUsername;
+    private String oldIdnPassword;
+    private String newIdnDBUrl;
+    private String newIdnUsername;
+    private String newIdnPassword;
+    private String newRegDBUrl;
+    private String newRegUsername;
+    private String newRegPassword;
+    private String enableDBMigrator;
+    private String enableConfigMigrator;
+    private String enableSyncMigrator;
+    private static final Logger log = Logger.getLogger(KeyRotationConfig.class);
+    private static final KeyRotationConfig instance = new KeyRotationConfig();
+
+    public static KeyRotationConfig getInstance() {
+
+        return instance;
+    }
+
+    /**
+     * Load the configurations placed in the properties.yaml file.
+     *
+     * @return KeyRotation config object.
+     * @throws KeyRotationException Exception thrown while loading the configs from properties.yaml file.
+     */
+    public KeyRotationConfig loadConfigs(String[] args) throws KeyRotationException {
+
+        String propertiesFilePath = Paths.get("components", "org.wso2.carbon.identity.keyrotation", "src",
+                "main", "resources", "properties.yaml").toString();
+        if (args.length > 0) {
+            propertiesFilePath = args[0];
+        }
+        File file = new File(propertiesFilePath);
+        if (!file.exists()) {
+            throw new KeyRotationException(
+                    "Error occurred, properties.yaml file not found in provided path, " + propertiesFilePath);
+        }
+        log.info("Loading Key Rotation Configs from path: " + propertiesFilePath);
+        KeyRotationConfig keyRotationConfig =
+                KeyRotationServiceUtils.loadKeyRotationConfig(propertiesFilePath);
+        checkKeyRotationConfigs(keyRotationConfig);
+        log.info("Successfully loaded the config file.");
+        return keyRotationConfig;
+    }
+
+    /**
+     * To check whether the loaded configurations are valid and not null.
+     *
+     * @param keyRotationConfig Configuration data needed to perform the task.
+     * @throws KeyRotationException Exception thrown while checking for null values in the loaded properties.yaml file.
+     */
+    public void checkKeyRotationConfigs(KeyRotationConfig keyRotationConfig) throws KeyRotationException {
+
+        Field[] props = KeyRotationConfig.getInstance().getClass().getDeclaredFields();
+        try {
+            for (int i = 0; i < props.length - 2; i++) {
+                if (props[i].get(keyRotationConfig) == null) {
+                    throw new KeyRotationException(
+                            "Error occurred, null value found in property, " + props[i].getName());
+                }
+                if ("oldISHome".equals(props[i].getName()) || "newISHome".equals(props[i].getName())) {
+                    File file = new File(props[i].get(keyRotationConfig).toString());
```

Comment:
Need to check the logic here

Reply:
This is to check whether the newISHome and oldISHome user provided file paths exist